source/shared/cpp/ObjectModel/InternalId.h

```diff
@@ -17,7 +17,7 @@ namespace AdaptiveCards
         static InternalId Current();
         static constexpr unsigned int Invalid = 0;
 
-        std::size_t Hash() const { return std::hash<unsigned int>()(m_internalId); }
```

Comment:
The instigating change here was that i needed to expose internal id out of the public API surface and i declared it as UINT32. When you build this for x64 you get errors assigning tying to assign a (64 bit) size_t to a UINT32, whereas a (32bit) unsigned int (which is what we're storing it in anyway) works fine. 

I removed the std::hash call because (a) it returned size_t and (b) the hash of the integer is itself. That said, I admit there might be a good reason to do this that i'm not aware of. @paulcam206 is there a reason this is a Bad Idea?